core/src/main.rs

```diff
@@ -73,7 +73,11 @@ async fn main() -> Result<(), CoordinatorError> {
     // ---------------------
 
     // Parse command line arguments
-    let args = config::parse_command_line_args().expect("error parsing command line args");
+    let args = tokio::task::spawn_blocking(config::parse_command_line_args)
```

Comment:
do we need to do this? it seems everything downstream of `parse_command_line_args` is synchronous